src/main.tsx

```diff
@@ -94,128 +94,48 @@ async function userIsMod({ reddit }: TriggerContext, userId: string, subredditNa
 }
 
 Devvit.addTrigger({
-  event: "PostCreate",
+  events: ["PostCreate", "CommentCreate"],
   async onEvent(event, context) {
     const { redis, reddit, settings } = context;
 
     console.log("Handling event:", JSON.stringify(event));
 
-    const authorId = event.author?.id;
-    if(!authorId) {
-      console.error("author ID missing");
-      return;
-    }
-
-    const contentId = event.post?.id;
-    if(!contentId) {
-      console.error("post ID missing");
-      return;
-    }
-
-    if(!event.subreddit?.name) {
-      console.error("subreddit name not present");
-      return;
-    }
-
-    if(await userIsMod(context, authorId, event.subreddit.name)) {
-      console.log("Ignoring moderator post:", contentId);
-      return;
-    }
-
-    const appSettings = await settings.getAll();
-
-    console.log("Settings:", JSON.stringify(appSettings));
-
-    const postLimit: number = appSettings["post-limit"] as number;
-    if(postLimit <= 0) {
-      return;
-    }
-
-    const key = `post:${authorId}`;
-
-    const windowStr = appSettings["post-window"] as string;
-
-    // window string to ms
-    const windowMs = parseDurationStr(windowStr, "ms");
-
-    const expiredBefore = new Date().getTime() - windowMs;
-
-    await redis.zRemRangeByScore(key, 0, expiredBefore);
-    const count = await redis.zCard(key);
-
-    if(count < postLimit) {
-      await redis.zAdd(key, {
-        member: contentId,
-        score: new Date().getTime(),
-      });
-
-      return;
-    }
-
-    const response = appSettings["post-reply"] as string;
-    if(!!response) {
-      try {
-        const comment = await reddit.submitComment({
-          id: contentId,
-          text: response,
-        });
-        comment.distinguish();
-      } catch(e) {
-        if(e instanceof Error) {
-          console.error(e.message);
-        } else {
-          console.error("unknown error submitting comment:", JSON.stringify(e));
-        }
-      }
-      
-    }
-
-    await reddit.remove(contentId, false);
-  },
-});
-
-Devvit.addTrigger({
-  event: "CommentCreate",
-  async onEvent(event, context) {
-    const { redis, reddit, settings } = context;
-
-    console.log("Handling event:", JSON.stringify(event));
+    const type = event.type === "PostCreate" ? "post" : "comment";
 
     const authorId = event.author?.id;
-    if(!authorId) {
+    if (!authorId) {
       console.error("author ID missing");
       return;
     }
 
-    const contentId = event.comment?.id;
-    if(!contentId) {
-      console.error("comment ID missing");
+    const contentId = event.type === "PostCreate" ? event.post?.id : event.comment?.id;
```

Comment:
Maybe

```typescript
const contentId = event[type]?.id
```

Reply:
I could't get that to work. The closest I've gotten is
```ts
import { CommentCreate, PostCreate } from '@devvit/protos';
const contentId = (event as PostCreate & CommentCreate)[type]?.id;
```

Reply:
This also works without needing to import anything: `const contentId = (event as any)[type]?.id as string;`.

Reply:
Can you set it in the method signature?

```typescript
async onEvent(event: (PostCreate & CommentCreate), context) {
```

Reply:
No, it has to be `&` not `|`, and `type` can't be determined in that case.

Reply:
found another solution that doesn't require imports: `const contentId = event[type as keyof Omit<typeof event, 'type'>]?.id;`

Reply:
Here's a simple example of the problem
```ts
type A = {
  type: 'A',
  x: string
};
type B = {
  type: 'B',
  y: string
};
function getProperty(item: A | B): string {
  const property = item.type === 'A' ? 'x' : 'y';
  return item[property];
}
```
And the error message on `item[property]`
> Element implicitly has an 'any' type because expression of type '"x" | "y"' can't be used to index type 'A | B'.
> Property 'x' does not exist on type 'A | B'. ts(7053)

Reply:
Makes me think that we should be able to infer the event type from the values provided to `events`. Maybe that'll be an upstream change in the package

Reply:
Could this be merged for now, and any further improvements added later?